syntax_and_semantics/cross-compilation.md

```diff
@@ -2,33 +2,60 @@
 
 Crystal supports a basic form of [cross compilation](http://en.wikipedia.org/wiki/Cross_compiler).
 
-In order to achieve this, the compiler executable provides two flags:
+## Preparation
+
+Cross-compilation will depend on `libcrystal` being present on the target system.  It can be built by following these steps:
+* Download the Crystal source code on the target machine.
+* Install the LLVM/compiler toolchain. (e.g. `sudo apt-get install clang` on Ubuntu)
```

Comment:
This isn't required to build libcrystal. It is however required for cross-compiling the compiler. Installing clang is a terrible way to install llvm.

I suggest this section is changed to make it clear installing LLVM/required libraries is only neccesary when cross-compiling the compiler. Libcrystal is neccesary regardless.

Reply:
Happy to change this to installing `llvm-3.x` directly.  Not sure I agree about the second part, given that this page is specifically in the context of cross-compiling already.  I think this speaks to the fact that the documentation on this page is a cross between general information about cross-compiling mixed with the specifics of cross-compiling the compiler.  Potentially worth separating these a little?

Reply:
That was largely what I was getting at: instructions for cross-compiling hello world should be covered completely *then* we should cover cross-compiling the compiler.

Reply:
I wish someone could port libcrystal to c bindings. It's just one function (though a difficult one because of the bindings)

Reply:
Luckily its not required for windows.

Reply:
Definitely agree it should be separated, I was hoping to improve it incrementally rather than all in one go.